pysd/py_backend/builder.py

```diff
@@ -65,13 +65,30 @@ def build(elements, subscript_dict, namespace, outfile_name):
     Python model "%(outfile)s"
     Translated using PySD version %(version)s
     """
-    from __future__ import division
-    import numpy as np
-    import xarray as xr
-    from pysd import utils, functions, external
-    from pysd import cache, subs
-    import os
+    from os import path\n'''
+    # intelligent import of needed functions and packages
+    if "numpy" in external_lib:
+        text += "    import numpy as np\n"
+    if "xarray" in external_lib:
+        text += "    import xarray as xr\n"
+    text += "\n"
+    if build_functions:
+        text += "    from pysd.py_backend.functions import %(methods)s\n"\
```

Comment:
Importing just the methods used is nice, and makes the code a little cleaner as well, when these functions are called. One potential issue is that we could run into namespace conflicts if a function that is imported shares the same name as a variable. In theory, this should be prevented by vensim - I don't think you can make a variable name that overwrites a builtin function. But, I'm not sure I completely trust that as a protection mechanism. Encapsulating in the `functions` namespace may be a bit more robust.

I don't have a strong opinion, though. @enekomartinmartinez, @alexprey, are you comfortable trusting vensim to prevent namespace conflicts between functions and variables?

Reply:
Hi @JamesPHoughton . No, I don't trust Vensim, that's why I made some changes in the `make_python_identifier` function from _utils.py_. First, I import the namespace of functions, external and decorators in the utils package, then I use the methods we are defining to complete the list of reserved_words and avoid having a model function with the same name as a PySD function.

Reply:
great solution! Thanks for explaining it to me. =)